src/qs_resp.F

```diff
@@ -226,6 +229,15 @@ SUBROUTINE resp_fit(qs_env)
          CALL print_fitting_points(qs_env, resp_env)
          CALL print_pot_from_resp_charges(qs_env, resp_env, particles, natom, output_unit)
 
+         ! In case of density functional embedding we need to save  the charges to qs_env
+         NULLIFY (dft_control)
+         CALL get_qs_env(qs_env, dft_control=dft_control)
+         IF (dft_control%qs_control%ref_embed_subsys) THEN
+            ALLOCATE (rhs_to_save(SIZE(resp_env%rhs)))
```

Comment:
you could possible use `ALLOCATE(rhs_to_save, SOURCE=resp_env%rhs)` instead of the two-step initialization